include/podio/ROOTDataSource.h

```diff
@@ -0,0 +1,117 @@
+#ifndef PODIO_DATASOURCE_H__
+#define PODIO_DATASOURCE_H__
+
+// STL
+#include <vector>
+#include <string>
+
+// ROOT
+#include <ROOT/RDataFrame.hxx>
+#include <ROOT/RDataSource.hxx>
+
+// Podio
+#include <podio/Frame.h>
+#include <podio/ROOTReader.h>
+#include <podio/CollectionBase.h>
+
+namespace podio {
+  using Record_t = std::vector<void*>;
+
+  class ROOTDataSource : public ROOT::RDF::RDataSource {
+    public:
+      explicit ROOTDataSource(const std::string& filePath, int nEvents = -1);
+      explicit ROOTDataSource(const std::vector<std::string>& filePathList,
+                              int nEvents = -1);
+
+      void SetNSlots(unsigned int nSlots) override;
+
+      template<typename T>
+      std::vector<T**> GetColumnReaders(std::string_view columnName);
```

Comment:
If I understand the documentation of `RDataSource` correctly, we don't need this at all. Also this is not overriding anything from the `RDataSource` base class. I think all of the "magic" is happening in the `GetColumnReadersImpl`, right?